timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/LockAcquirer.java

```diff
@@ -26,7 +26,7 @@
 import com.google.common.base.Throwables;
 import com.palantir.logsafe.SafeArg;
 
-public class LockAcquirer {
+public class LockAcquirer implements AutoCloseable {
```

Comment:
kinda weird to make `LockAcquirer` closeable here. Is it not sufficient to just close the executor like `reaperExecutor` already is? Since `AsyncLockService` "owns" the executor so to speak.

Reply:
No strong preference - reason I went with this is that `AsyncLockService` currently doesn't maintain any references to `timeoutExecutor`